src/scenario/action_types/trade_route_amount.c

```diff
@@ -0,0 +1,34 @@
+#include "trade_route_amount.h"
+
+#include "city/message.h"
+#include "empire/city.h"
+#include "empire/trade_route.h"
+
+int scenario_action_type_trade_route_amount_execute(scenario_action_t *action)
+{
+    int32_t route_id = action->parameter1;
+    int32_t resource = action->parameter2;
+    int32_t amount = action->parameter3;
+    int32_t show_message = action->parameter4;
+
```

Comment:
You should check if the `route_id` or the `resource` are valid before executing this, otherwise this may end up crashing the game due to invalid data.

Maybe when there's a problem you should display a warning in the log.

Reply:
Done.